### PR TITLE
DEV-4427 fix submission back button

### PR DESCRIPTION
**Pull request by ebdabbs**
**High level description:**

Refactors the submission container and submission pages to use updated React router features, improve the architecture, and fixes a bug preventing users from navigating back to the submissions table via the browser's back button.

**Technical details:**

- Consolidates layout styling into a single `usa-da-submission-page` class
- Uses React router `Link` elements for the progress bar and Next buttons
- Renamed `AddDataHeader` to `SubmissionHeader` and make the API request in the container, pass the data via props
- Separated the routes `/submission/:submissionID` and `/submission/:submissionID/:step` to correspond to different containers, so that the sole purpose of `SubmissionContainer` is to request the original step and redirect to its corresponding url. `SubmissionStepsContainer` stores information about the current submission in state, handles and validates changes to the step via url, and requests metadata. 
- Makes an API call for the maximum current step number each time the url `:step` parameter changes, and prevents users from accessing invalid steps

**Link to JIRA Ticket:**

[DEV-4427](https://federal-spending-transparency.atlassian.net/browse/DEV-4427)

The following are ALL required for the PR to be merged:

Author: 
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled Demo including Testing/Front-end
- [x] Verified cross-browser compatibility
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)

Reviewer(s):
- [x] Frontend review completed

### CODE REVIEW

File: src/js/components/submission/SubmissionPage.jsx [hunk 1]
```diff
@@ -1,104 +1,81 @@
...
+                break;
+            case 5:
+                content = <ReviewDataContainer submissionID={submissionID} errorFromStep={this.props.errorFromStep} />;
+                break;
+            default:
+                content = null;
+        }
+
+        if (loading) content = this.loadingMessage();
+        if (error) content = this.errorMessage();
```
Comment (alburde1) [inline src/js/components/submission/SubmissionPage.jsx]: If these are already `if` statements, do we need the functions at all? Can we just do
```if (loading) content = <ReviewLoading />```
for example?

File: src/js/components/submission/SubmissionHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/**
+* SubmissionHeader.jsx
+* Created by Kyle Fox 2/19/16
```
[the remark this answers is not in the document]
Comment (ebdabbs) [reply src/js/components/submission/SubmissionHeader.jsx]: It was renamed from `AddDataHeader`, but it's been changed enough that I can update it 😂

File: src/_scss/layouts/validation/progress/_step.scss [hunk 3]
```diff
@@ -1,85 +1,95 @@
...
+        display: block;
+        text-align: center;
+        opacity: 0.3;
+		font-size: 1.2rem;
+		a {
+			color: $color-gray-dark;
+			&:hover {
+				text-decoration: none;
+			}
+        };
```
[the remark this answers is not in the document]
Comment (ebdabbs) [reply src/_scss/layouts/validation/progress/_step.scss]: Screen Shot 2020-03-05 at 1 30 01 PM
It looks normal in my editor.. may be just weirdness with the GitHub view?

[the remark this answers is not in the document]
Comment (alburde1) [reply src/_scss/layouts/validation/progress/_step.scss]: It may be tabs vs spaces?

**Pull request merged**